com.unity.render-pipelines.core/Runtime/XR/XRSystem.cs

```diff
@@ -0,0 +1,377 @@
+using System;
+using System.Collections.Generic;
+
+#if ENABLE_VR && ENABLE_XR_MODULE
+using UnityEngine.XR;
+#endif
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// Used by render pipelines to communicate with XR SDK.
+    /// </summary>
+    public static class XRSystem
```

Comment:
Rather than removing XRSystem from URP and forcing both URP and HDRP to use the same XRSystem class, we should allow XR to have different features that are on URP vs HDRP. HDRP and URP have different needs, with HDRP more console and PC-centric while URP is more mobile-centric.

Rather than a single XRSystem class, we can create XR base class in Core that has basic common features. URP can subclass from XR base class to allow for customizations and experimental features geared toward Quest/mobile.

Specifically for XRSystem:

1. Create a base class XRBaseSystem in core and let XRSystem in each pipeline subclass from XRBaseSystem.

2. This allows experimental features such as late latching to first populate and be developed in URP (adding to URP's XRSystem). For late latching, HDRP won't have any immediate need for late latching given PC targets at 90Hz with a much faster GPU. Oculus also has no plans to support late latching for Rift.

3. There are other differences between HDRP (console/PC) and URP (mobile) such as the need for memoryless textures and Vulkan subpasses on mobile, that allowing subclassing of XRBaseSystem will allow more flexible render pipeline features.

Reply:
I think it's worth considering this design as a second step, if we encounter a good use case.
But for now, with this class being static, it's easy to handle the differences directly in the respective render pipeline.
Adding inheritance right now will complexify the API and will make cross-pipeline code harder to deal with.

Reply:
Here's a good use case. The new late latching for URP adds several functions to XRSystem and adds late latching states to XRPass. These shouldn't go into core pipeline since late latching isn't supported on HDRP nor needed in HDRP as I mentioned above. 
If each pipeline can subclass from a common XRBaseSystem, new XR features like late latching can exist in one pipeline without polluting other pipelines.

Reply:
Any thoughts on my above comments? By merging in late latching and moving XRSystem to Core, late latching is exposed to HDRP and any user defined SRP's, which we don't support or tested with late latching. 

There's also late latching code changed during the merge, which needs to be dealt with (I'll do a comparison).